pkg/vm/engine/disttae/logtailreplay/partition_state.go

```diff
@@ -400,7 +400,29 @@ func (p *PartitionState) HandleMetadataInsert(ctx context.Context, input *api.Ba
 				}
 				iter.Release()
 			}
-
+			if !entryStateVector[i] && len(entry.DeltaLoc) != 0 {
```

Comment:
Why not add these conditions to line 380? Seems exactly the same.

Reply:
Done.